pkg/util/label.go

```diff
@@ -124,3 +125,20 @@ func RecordManagedLabels(object *unstructured.Unstructured) {
 	annotations[workv1alpha2.ManagedLabels] = strings.Join(managedKeys, ",")
 	object.SetAnnotations(annotations)
 }
+
+// DedupeAndMergeFinalizers merges the new finalizers into exist finalizers.
+func DedupeAndMergeFinalizers(existFinalizers, newFinalizers []string) []string {
+	if existFinalizers == nil {
+		return existFinalizers
+	}
```

Comment:
I don't understand why it needs a return here, but I don't think it should be.
Consider one situation where len(existFinalizer) != 0 and len(newFinalizers) > 0.